packages/react-hooks/src/useEventListener.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright IBM Corp. 2018, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect, useRef } from 'react';
+
+/**
+ * Given a reference to an HTMLElement, specify an event and callback to invoke
+ * whenever the event occurs. Optionally, you can specify options for the event
+ * handler.
+ *
+ * @example
+ * function MyComponent() {
+ *   useEventListener(window, 'click', () => {
+ *     // Window was clicked
+ *   });
+ * }
+ *
+ * @param {HTMLElement} element - the element for which the event occurs
+ * @param {string} name - the name of the event
+ * @param {Function} callback - the callback invoked by the listener
+ * @param {object?} options - configuration object for the listener
+ */
+export function useEventListener(element, name, callback, options) {
+  const savedCallback = useRef(callback);
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  useEffect(() => {
+    if (!element || !element.addEventListener) {
```

Comment:
Does this mean someone would have to pass in a dom node? Would it be better to use a ref? Probably don't want to encourage folks digging through the dom.

Reply:
@vpicone the idea was that if you wanted to use a ref you'd pass in `ref.current` to the hook and if the ref wasn't initialized yet it would hit the false-y check on `!element`

Reply:
Ah I see